### ISSUE TITLE
No slider with only 1 item allowed

### ISSUE THREAD

**Issue by JeroenReumkens** [ganlanyuan/tiny-slider#170]
__Issue description__:  
When there is only one slide, the slider won't initiate. What is the reasoning behind this? In a CMS it could sometimes be the case that they will only add one slide. Right now it will break because no width is applied (if have 4 items visible per page, so now 1 slide is as wide as 4) and the arrows are still visible. 

Because of this code: 
``` 
 if (options.container.children.length < 2) {
    if (supportConsoleWarn) { console.warn('Slides less than 2.'); }
    return;
   } 
```

My suggestion is to make it an option to initialise the slider with only one slide, since I can not really think of a reason not to? 

_Tiny-slider version_:    2.6.0
_Browser name && version_:  Chrome 64
_OS name && version_:   OSX 10.13.3

**Comment by ganlanyuan** [comment]
If all slides can be displayed on one screen, it doesn't make sense to make a slider.

**Comment by frontend-fyi** [comment]
That would only make sense if there is only 1 slide per page. Since this check doesn’t take into account the number of slides per page, I don’t think this is the intention of that code. 

For example; if you have 3 slides per page, with 2 or more slides the width of the items will be 33.33%. So,even if there are 2 slides. 
But if there is only one, instead of being 33.333%, it won’t have a width so it becomes 100%. 

Of course you can do things with the :only-child selector in css, but that would mean that you need to meantain your number of slides per page breakpoints both in css and js. 

That why I think this still is a bug.

**Comment by JeroenReumkens** [comment]
That would only make sense if there is only 1 slide per page. Since this check doesn’t take into account the number of slides per page, I don’t think this is the intention of that code.

For example; if you have 3 slides per page, with 2 or more slides the width of the items will be 33.33%. So,even if there are 2 slides.
But if there is only one, instead of being 33.333%, it won’t have a width so it becomes 100%.

Of course you can do things with the :only-child selector in css, but that would mean that you need to meantain your number of slides per page breakpoints both in css and js.

That why I think this still is a bug.

**Comment by ganlanyuan** [comment]
Now I understand what you mean, thanks.
Updated in [this commit](https://github.com/ganlanyuan/tiny-slider/commit/989aeb4ba5578f69a00dede69f041b2f8de943d7).

**Comment by JeroenReumkens** [comment]
Nice, that is exactly the fix! 
However, could you test what this does with 0 slides? It now continues and it looks like it breaks with zero slides. So maybe just change the line to < 1?

**Comment by ganlanyuan** [comment]
Right, thanks again!
Updated in [this commit](https://github.com/ganlanyuan/tiny-slider/commit/5673da91cb0c3a88b66f4bb47f5dab5475faeab8).

**Comment by iamricky** [comment]
I'm still getting this bug. My slider breaks when there's only one slide. Also, `autoplay` doesn't work for `<video>` inside tiny-slide, unless there's only one slide, then videos work but the slider breaks.

**Comment by ganlanyuan** [comment]
Fixed in v2.7.0
@iamricky regarding the video issue, could you open another ticket?
